File: Sources/ATTNItem.m
//
//  ATTNItem.m
//  attentive-ios-sdk
//
//  Created by Wyatt Davis on 12/16/22.
//

#import "ATTNItem.h"

NS_ASSUME_NONNULL_BEGIN

@implementation ATTNItem

- (instancetype)initWithProductId:(NSString *)productId productVariantId:(NSString *)productVariantId price:(ATTNPrice *)price {
  if (self = [super init]) {
    self->_productId = productId;
    self->_productVariantId = productVariantId;
    self->_price = price;
    self->_quantity = 1;
  }

  return self;
}

@end

NS_ASSUME_NONNULL_END

File: Sources/Migration/Public/Events/ATTNItem.swift
//
//  ATTNItem.swift
//  attentive-ios-sdk-framework
//
//  Created by Vladimir - Work on 2024-05-30.
//

import Foundation

@objc(ATTNItem)
public final class ATTNItem: NSObject {
  @objc public let productId: String
  @objc public let productVariantId: String
  @objc public let price: ATTNPrice
  @objc public var quantity: Int
  @objc public var productImage: String?
  @objc public var name: String?
  @objc public var category: String?

  @objc(initWithProductId:productVariantId:price:)
  public init(productId: String, productVariantId: String, price: ATTNPrice) {
    self.productId = productId
    self.productVariantId = productVariantId
    self.price = price
    self.quantity = 1
  }
}
